### ISSUE TITLE
Systemclt status shows state: degraded

### ISSUE THREAD

**Issue by michalgutowski** [oracle/vagrant-boxes#144]
Hi,

I noticed that whenever I'm doing vagrant up on OL7 boxes the systemctl command reports that overall system status is **degraded**:

```
[root@master ~]# systemctl status | head -5
● master.vagrant.vm
    State: degraded
     Jobs: 0 queued
   Failed: 1 units
    Since: Tue 2019-08-13 09:01:31 UTC; 11min ago
```
It looks like there is a problem with migrating selinux policies:

```
[root@master ~]# systemctl list-units --failed
  UNIT                                                  LOAD   ACTIVE SUB    DESCRIPTION
● selinux-policy-migrate-local-changes@targeted.service loaded failed failed Migrate local SELinux policy changes from the old store structure to the new stru

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.

1 loaded units listed. Pass --all to see loaded but inactive units, too.
To show all installed unit files use 'systemctl list-unit-files'.
[root@master ~]# systemctl status selinux* -l
● selinux-policy-migrate-local-changes@targeted.service - Migrate local SELinux policy changes from the old store structure to the new structure
   Loaded: loaded (/usr/lib/systemd/system/basic.target.wants/../selinux-policy-migrate-local-changes@.service; static; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2019-08-13 09:01:33 UTC; 21min ago
 Main PID: 2727 (code=exited, status=208/STDIN)

Aug 13 09:01:33 localhost.localdomain systemd[1]: Starting Migrate local SELinux policy changes from the old store structure to the new structure...
Aug 13 09:01:33 localhost.localdomain systemd[2727]: Failed at step STDIN spawning /usr/libexec/selinux/selinux-policy-migrate-local-changes.sh: Inappropriate ioctl for device
Aug 13 09:01:33 localhost.localdomain systemd[1]: selinux-policy-migrate-local-changes@targeted.service: main process exited, code=exited, status=208/STDIN
Aug 13 09:01:33 localhost.localdomain systemd[1]: Failed to start Migrate local SELinux policy changes from the old store structure to the new structure.
Aug 13 09:01:33 localhost.localdomain systemd[1]: Unit selinux-policy-migrate-local-changes@targeted.service entered failed state.
Aug 13 09:01:33 localhost.localdomain systemd[1]: selinux-policy-migrate-local-changes@targeted.service failed.
```

**Comment by michalgutowski** [comment]
Looking at the similar issue it seems the problem is with console definition in the grub command line - we are configuring ttyS0 in kernel parameters but our vm doesn't have such device specified.

`GRUB_CMDLINE_LINUX="no_timer_check console=tty0 console=ttyS0,115200 net.ifnames=0 biosdevname=0 crashkernel=auto rd.lvm.lv=vg_main/lv_root rd.lvm.lv=vg_main/lv_swap"`

> The crux of the problem is that, if you have `console=ttyS0,115200`, or some other ttySx serial console definition, as the last console on the kernel command line, the kernel will attempt to activate the serial console regardless of whether or not the serial port actually exists.

Because the `/lib/systemd/system/selinux-policy-migrate-local-changes\@.service` file contains `StandardInput=tty` it tries to do `ioctl()` on `/dev/console` which is pointing to ttyS0. That fails and as a result the selinux-policy-migrate-local-changes service fails.

**Resolution**
Change the order of console definitions in /etc/default/grub.

From:

`GRUB_CMDLINE_LINUX="no_timer_check console=tty0 console=ttyS0,115200 net.ifnames=0 biosdevname=0 crashkernel=auto rd.lvm.lv=vg_main/lv_root rd.lvm.lv=vg_main/lv_swap"`

To:

`GRUB_CMDLINE_LINUX="no_timer_check console=ttyS0,115200 console=tty0 net.ifnames=0 biosdevname=0 crashkernel=auto rd.lvm.lv=vg_main/lv_root rd.lvm.lv=vg_main/lv_swap"`

Or completely remove the ttyS0 definition:

`GRUB_CMDLINE_LINUX="no_timer_check console=tty0 net.ifnames=0 biosdevname=0 crashkernel=auto rd.lvm.lv=vg_main/lv_root rd.lvm.lv=vg_main/lv_swap"`

Apply changes and restart:

```
# grub2-mkconfig -o /boot/grub2/grub.cfg
# reboot
```

**Comment by totalamateurhour** [comment]
Thanks @michalgutowski, looking into this

**Comment by totalamateurhour** [comment]
Fixed in OL 7.7 box.
